### PR TITLE
Uses exclusive attachment for IAM roles and simplifies logic for merging policy documents

### PR DESCRIPTION
**Pull request by lorengordon**

### CODE REVIEW

**Comment by lorengordon** [comment]
results of `make test`:
```
PASS
ok      tardigarde-ci/tests     328.125s
[terratest/test]: Completed successfully!
```

File: modules/policy_document/variables.tf [hunk 1]
```diff
@@ -4,12 +4,8 @@ variable "template" {
 }
 
 variable "template_paths" {
-  description = "List of directory paths to search for the `template` policy document. Max of 10 paths. Policy statements are merged by SID in list order"
+  description = "List of directory paths to search for the `template` policy document. Policy statements are merged by SID in list order. If the `template` does not exist at a given template_path, that path is ignored"
```
Comment (eemperor) [inline modules/policy_document/variables.tf:7]: Maybe more clear to mention empty policy is inserted?

```suggestion
  description = "List of directory paths to search for the `template` policy document. Policy statements are merged by SID in list order. If the `template` does not exist at a given template_path, an empty policy is inserted"
```

**Review by eemperor — COMMENTED** [review]
@lorengordon I just have the one minor comment about the updated description for the `template_paths`.  Looks good overall.

File: modules/policy_document/variables.tf (hunk 1 above)
Comment (lorengordon) [reply modules/policy_document/variables.tf:7]: That probably makes sense, considering if the template doesn't exist at _any_ template_paths, then the result is an empty policy. Would helps explain _why_ that occurs... 😂

Comment (lorengordon) [reply modules/policy_document/variables.tf]: Done, re-running tests...

**Comment by lorengordon** [comment]
results of `make test`:
```
PASS
ok      tardigarde-ci/tests     285.031s
[terratest/test]: Completed successfully!
```

**Review by eemperor — APPROVED** [review]
🚀

**Review by kumoy — APPROVED** [review]

**Pull request merged**